crates/sui-core/src/transaction_input_checker.rs

```diff
@@ -19,8 +19,8 @@ use sui_types::{
 };
 use tracing::instrument;
 
-async fn get_gas_status(
-    store: &AuthorityStore,
+pub async fn get_gas_status(
```

Comment:
I am in the process of changing this for gas smashing, and things are (at the moment at least) close to what you have rather that what it was.
In any case is there a reason to remove the store from the API?
is that related to the fact we have fetched the objects already and we do not want to do it again?

Reply:
Yes lets not have a dependency on the store: 
(1) As I do in this PR we might want to have execution without access to a full store (and its trivial besides for this gas call).
(2) We only want to be reading things once indeed, any calls that iteratively read values from the store should eventually be eliminated. Its easier if they are not all over the code.

Reply:
Talked to @tzakian about this too -- did not realize you were touching this code.